### PR TITLE
Docs: Add shading information to FAQ

### PR DESCRIPTION
**Pull request by pomadchin**
A very popular question about Guava versions incompatibilities, this PR clarifies a bit the situation and provides a solution.

### CODE REVIEW

**Comment by fosskers** [comment]
Do you think this would be better placed in our FAQ, as opposed to the Setup? I'm concerned that users would perceive it as "GeoTrellis is hard to even get started".

**Comment by pomadchin** [comment]
@fosskers I agree with you, let's move to FAQ

**Comment by fosskers** [comment]
How do Guava conflicts usually manifest? I know they've happened to me before, but I feel like they don't come straight out and say "java.lang.Exception: Your guava situation is crazy, bro".

**Comment by pomadchin** [comment]
@fosskers :D they happened to you more times than you even remember.

**Comment by fosskers** [comment]
I know :( Runtime error heaven over here.

File: docs/guide/faq.rst [hunk 1]
```diff
@@ -158,3 +158,41 @@ This does.
...
+Full possible exception message:
+
+.. code::
+
+    Caused by: java.lang.IllegalStateException: Detected Guava issue #1635 which indicates that a version of Guava less
+    than 16.01 is in use.  This introduces codec resolution issues and potentially other incompatibility issues in the driver.
+    Please upgrade to Guava 16.01 or later.
+
+
+GeoTrellis depends on a huge number of complex dependencies that may cause a dependency hell. One of such dependency
```
Comment (fosskers) [inline docs/guide/faq.rst]: `a dependency hell` -> `dependency hell`

File: docs/guide/faq.rst [hunk 2]
```diff
@@ -158,3 +158,41 @@ This does.
...
+is a guava library. ``GeoTrellis ETL`` and ``GeoTrellis Cassandra`` depend on ``Guava 16.01``, but Hadoop depends on ``Guava 11.0.2``
```
Comment (fosskers) [inline docs/guide/faq.rst]: `is a guava library` -> `is the Guava library`

File: docs/guide/faq.rst [hunk 3]
```diff
@@ -158,3 +158,41 @@ This does.
...
+that causes strong libs incompatibility runtime issues. When two different versions of the same library both available in a
```
Comment (fosskers) [inline docs/guide/faq.rst]: `that causes strong libs...` -> `which causes runtime issues due to library incompatibility`

`the same library both available` -> `the same library are both available`

File: docs/guide/faq.rst [hunk 4]
```diff
@@ -158,3 +158,41 @@ This does.
...
+Spark classpath and in a fat assembly jar, Spark would use library version from its classpath.
```
Comment (fosskers) [inline docs/guide/faq.rst]: `in a Spark classpath` -> `in the Spark classpath`

`Spark would use` -> `Spark will use`

File: docs/guide/faq.rst [hunk 5]
```diff
@@ -158,3 +158,41 @@ This does.
...
+
+There're two possible solutions:
```
Comment (fosskers) [inline docs/guide/faq.rst]: `There're` -> `There are`

**Pull request merged**